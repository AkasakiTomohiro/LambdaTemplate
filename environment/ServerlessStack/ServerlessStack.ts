import { Stack, Construct, StackProps } from '@aws-cdk/core';
import { ServerlessFunctionRole } from './Role/ServerlessFunctionRole';
import { ServerlessFunction } from './Lambda/ServerlessFunction';

export class ServerlessStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const serverlessFunctionRole = ServerlessFunctionRole(this);

    ServerlessFunction(this, serverlessFunctionRole);
  }
}
