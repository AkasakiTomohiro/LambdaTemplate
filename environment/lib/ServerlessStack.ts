import { Role, ServicePrincipal } from '@aws-cdk/aws-iam';
import { CfnFunction } from '@aws-cdk/aws-sam';
import { Stack, Construct, StackProps } from '@aws-cdk/core';

export class ServerlessStack extends Stack {
  /**
   * Serverless Function
   */
  public serverlessFunction: CfnFunction;

  /**
   * Serverless Function„ÅÆRole
   */
  public serverlessFunctionRole: Role;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    this.serverlessFunctionRole = new Role(this, 'ServerlessFunctionRole', {
      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
      managedPolicies: [
        { managedPolicyArn: 'arn:aws:iam::aws:policy/CloudWatchFullAccess' },
        { managedPolicyArn: 'arn:aws:iam::aws:policy/CloudWatchLogsFullAccess' },
        { managedPolicyArn: 'arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess' }
      ]
    });

    this.serverlessFunction = new CfnFunction(this, 'ServerlessFunction', {
      runtime: 'nodejs10.x',
      codeUri: './dist',
      handler: 'index.handler',
      memorySize: 128,
      timeout: 8,
      role: this.serverlessFunctionRole.roleArn,
      events: { GetResource: { type: 'Api', properties: { method: 'any', path: '/{proxy+}' } } }
    });
  }
}
