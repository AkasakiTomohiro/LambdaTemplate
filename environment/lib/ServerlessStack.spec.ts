import { expect as expectCDK, matchTemplate, MatchStyle, SynthUtils } from '@aws-cdk/assert';
import { ServerlessStack } from './ServerlessStack';
import { App } from '@aws-cdk/core';

test('Stack', () => {
  const app = new App();
  const stack = new ServerlessStack(app, 'ServerlessStack');

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
