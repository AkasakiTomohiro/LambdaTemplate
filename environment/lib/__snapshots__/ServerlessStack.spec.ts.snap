// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack 1`] = `
Object {
  "Resources": Object {
    "ServerlessFunction": Object {
      "Properties": Object {
        "CodeUri": "./dist",
        "Events": Object {
          "GetResource": Object {
            "Properties": Object {
              "Method": "any",
              "Path": "/{proxy+}",
            },
            "Type": "Api",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "ServerlessFunctionRoleD8A50BCD",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
        "Timeout": 8,
      },
      "Type": "AWS::Serverless::Function",
    },
    "ServerlessFunctionRoleD8A50BCD": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/CloudWatchFullAccess",
          "arn:aws:iam::aws:policy/CloudWatchLogsFullAccess",
          "arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess",
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Transform": "AWS::Serverless-2016-10-31",
}
`;
